tests/test_cli_compile.py

```diff
@@ -2582,6 +2582,31 @@ def test_input_formats(fake_dists, runner, make_module, fname, content):
     assert "extra ==" not in out.stderr
 
 
+def test_error_in_pyproject_toml(fake_dists, runner, make_module, capfd):
+    """
+    Test that an error in pyproject.toml is reported.
+    """
+    fname = "pyproject.toml"
+    invalid_content = dedent(
+        """\
+        [project]
+        invalid = "metadata"
+        """
+    )
+    meta_path = make_module(fname=fname, content=invalid_content)
+    out = runner.invoke(
+        cli, ["-n", "--no-build-isolation", "--find-links", fake_dists, meta_path]
+    )
+    assert out.exit_code == 2, out.stderr
+    assert "`project` must contain ['name'] properties" not in capfd.readouterr().err
+    out = runner.invoke(
+        cli, ["-n", "--no-build-isolation", "-v", "--find-links", fake_dists, meta_path]
+    )
+    assert out.exit_code == 2, out.stderr
+    captured = capfd.readouterr()
+    assert "`project` must contain ['name'] properties" in captured.err
```

Comment:
We typically prefer one action per test. As this test performs two actions, it would be ideally to split this into two separate tests using parametrization, for example:

```python
@pytest.parametrize("verbose", (True, False))
def test_error_in_pyproject_toml(..., verbose):
    ...
    options = ["-n", "--no-build-isolation", "--find-links", fake_dists, meta_path]
    if verbose:
        options.append("--verbose")
    out = runner.invoke(cli, options)
    ...
    assert ("`project` must contain ['name'] properties" in captured.err) is verbose
```

Reply:
Implemented. Thanks for the suggestion!